packages/core/helper-plugin/lib/src/hooks/useFieldHint/index.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import { getFieldUnits, getMinMax } from './utils';
+
+/**
+ * @description
+ * A hook for generating the hint for a field
+ * @type {
+ * ({ description: { id: string, defaultMessage: string },
+ *    type: string,
+ *    fieldSchema: { minLength?: number|string; maxLength?: number|string; max?: number|string; min?: number|string }
+ * })
+ * => { hint: ''|Array }
+ * }
+ */
+const useFieldHint = ({ description, fieldSchema, type }) => {
+  const { formatMessage } = useIntl();
+  const [hint, setHint] = useState([]);
+
+  /**
+   * @returns {String}
+   */
+  const buildDescription = useCallback(
+    (desc) =>
+      desc?.id
+        ? formatMessage({ id: desc.id, defaultMessage: desc.defaultMessage }, { ...desc.values })
+        : '',
+    [formatMessage]
+  );
+
+  /**
+   * @returns {''|Array}
+   */
+  const buildHint = useCallback(
+    (desc) => {
+      const { maximum, minimum } = getMinMax(fieldSchema);
+      const units = getFieldUnits({
+        type,
+        minimum,
+        maximum,
+      });
+
+      const minIsNumber = typeof minimum === 'number';
+      const maxIsNumber = typeof maximum === 'number';
+      const hasMinAndMax = maxIsNumber && minIsNumber;
+      const hasMinOrMax = maxIsNumber || minIsNumber;
+
+      if (!desc?.id && !hasMinOrMax) {
+        return '';
+      }
+
+      return formatMessage(
+        {
+          id: 'content-manager.form.Input.hint.text',
+          defaultMessage:
+            '{min, select, undefined {} other {min. {min}}}{divider}{max, select, undefined {} other {max. {max}}}{unit}{br}{description}',
+        },
+        {
+          min: minimum,
+          max: maximum,
+          description: buildDescription(desc),
+          unit: units?.message && hasMinOrMax ? formatMessage(units.message, units.values) : null,
+          divider: hasMinAndMax
+            ? formatMessage({
+                id: 'content-manager.form.Input.hint.minMaxDivider',
+                defaultMessage: ' / ',
+              })
+            : null,
+          br: hasMinOrMax ? <br /> : null,
+        }
+      );
+    },
+    [formatMessage, buildDescription, fieldSchema, type]
+  );
+
+  useEffect(() => {
+    setHint(buildHint(description));
```

Comment:
Any downside to just returning the return value of buildHint? Feels unnecessary - would reduce the useCallbacks too. Wdyt?

Reply:
Nope that's a good idea thanks! Fixed in ea95c70ae3a36533de781858caff38a6592ab115